lib/pbench/test/unit/server/test_template.py

```diff
@@ -0,0 +1,88 @@
+import datetime
+import os
+from pathlib import Path
+from posix import stat_result
+from stat import ST_MTIME
+import pytest
+
+from pbench.server.database.models.template import (
+    Template,
+    TemplateNotFound,
+)
+
+
+@pytest.fixture(autouse=True)
+def fake_mtime(monkeypatch):
+    """
+    Template's init event listener provides the file's modification date to
+    support template version control. For unit testing, mock the stat results
+    to appear at a fixed time.
+
+    Args:
+        monkeypatch ([type]): [description]
+    """
+
+    def fake_stat(file: str):
+        """
+        Create a real stat_result using an actual file, but change the st_mtime
+        to a known value before returning it.
+
+        Args:
+            file: filename (not used)
+
+        Returns:
+            mocked stat_results
+        """
+        s = os.stat(".")
+        t = int(datetime.datetime(2021, 1, 29, 0, 0, 0).timestamp())
+        f = list(s)
+        f[ST_MTIME] = t
+        return stat_result(f)
+
+    with monkeypatch.context() as m:
+        m.setattr(Path, "stat", fake_stat)
+        yield
+
+
+class TestTemplate:
+    def test_construct(self):
+        """ Test dataset contructor
+        """
+        template = Template(
+            name="run",
+            template_name="tname",
+            file="run.json",
+            index_template="run.{year}-{month}",
+            settings={"none": False},
+            mappings={"properties": None},
+            version=5,
+        )
+        template.add()
+        assert template.name == "run"
+        assert template.mtime == datetime.datetime(2021, 1, 29, 0, 0, 0)
```

Comment:
You should have a symbolic name for `datetime.datetime(2021, 1, 29, 0, 0, 0)` since you use it in more than one place and expect it to have the same value.

Reply:
I believe I use it once in the fixture and once in a test; I considered using the same constant, but I actually like having the test check against the literal rather than blindly referencing the mock, as a consistency double-check.